processing/src/test/java/io/druid/query/metadata/SegmentMetadataUnionQueryTest.java

```diff
@@ -0,0 +1,143 @@
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.aggregation.DoubleMaxAggregatorFactory;
+import io.druid.query.aggregation.DoubleMinAggregatorFactory;
+import io.druid.query.aggregation.PostAggregator;
+import io.druid.query.metadata.metadata.ColumnAnalysis;
+import io.druid.query.metadata.metadata.ListColumnIncluderator;
+import io.druid.query.metadata.metadata.SegmentAnalysis;
+import io.druid.query.metadata.metadata.SegmentMetadataQuery;
+import io.druid.segment.IncrementalIndexSegment;
+import io.druid.segment.QueryableIndexSegment;
+import io.druid.segment.TestHelper;
+import io.druid.segment.TestIndex;
+import io.druid.segment.column.ValueType;
+import org.joda.time.DateTime;
+import org.joda.time.Interval;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RunWith(Parameterized.class)
+public class SegmentMetadataUnionQueryTest
+{
+  private static final QueryRunnerFactory FACTORY = new SegmentMetadataQueryRunnerFactory(
+      new SegmentMetadataQueryQueryToolChest(new SegmentMetadataQueryConfig()),
+      QueryRunnerTestHelper.NOOP_QUERYWATCHER
+  );
+  private final QueryRunner runner;
+  private final boolean mmap;
+
+  public SegmentMetadataUnionQueryTest(
+      QueryRunner runner,
+      boolean mmap
+  )
+  {
+    this.runner = runner;
+    this.mmap = mmap;
+  }
+
+  @Parameterized.Parameters
+  public static Iterable<Object[]> constructorFeeder() throws IOException
+  {
+    return ImmutableList.of(
+        new Object[]{
+            QueryRunnerTestHelper.makeUnionQueryRunner(
+                FACTORY,
+                new QueryableIndexSegment(
+                    QueryRunnerTestHelper.segmentId,
+                    TestIndex.getMMappedTestIndex()
+                )
+            ), true
+        },
+        new Object[]{
+            QueryRunnerTestHelper.makeUnionQueryRunner(
+                FACTORY,
+                new IncrementalIndexSegment(
+                    TestIndex.getIncrementalTestIndex(),
+                    QueryRunnerTestHelper.segmentId
+                )
+            ), false
+        }
+    );
+  }
+
+
+  @Test
+  public void testSegmentMetadataUnionQuery()
+  {
+    System.out.println();
+    SegmentAnalysis expected = new SegmentAnalysis(
+        QueryRunnerTestHelper.segmentId,
+        Lists.newArrayList(new Interval("2011-01-12T00:00:00.000Z/2011-04-15T00:00:00.001Z")),
+        ImmutableMap.of(
+            "placement",
+            new ColumnAnalysis(
+                ValueType.STRING.toString(),
+                false,
+                mmap ? 43524 : 43056,
+                1,
+                "preferred",
+                "preferred",
+                null
+            )
+        ),
+        mmap ? 287928 : 291020,
+        4836,
```

Comment:
Is this value right for the union dataSource? (hard to see since there's no non-union test to compare it to)